File: src/main/java/org/bakkes/game/controller/event/input/CompositeKeyListener.java
package org.bakkes.game.controller.event.input;

import java.util.Collection;

import com.google.inject.Inject;

public class CompositeKeyListener implements IKeyListener{

	private @Inject Collection<IKeyListener> keyListeners;
	@Override
	public void KeyDown(final Key key) {
		keyListeners.forEach(k -> k.KeyDown(key));
	}

	@Override
	public void KeyUp(final Key key) {
		keyListeners.forEach(k -> k.KeyUp(key));
	}

	public void clear(){
		this.keyListeners.clear();
	}
	public boolean add(final IKeyListener listner){
		return this.keyListeners.add(listner);
	}
}

File: src/main/scala/org/bakkes/game/controller/event/input/CompositeKeyListener.scala
package org.bakkes.game.controller.event.input

import java.util.Collection
import scala.collection.JavaConversions._

import com.google.inject.Inject

class CompositeKeyListener @Inject() (listeners : Collection[IKeyListener]) extends IKeyListener{

	override def KeyDown(key:Key) = {
		listeners.foreach{_.KeyDown(key)}
	}

	override def KeyUp(key:Key) = {
		listeners.foreach{_.KeyUp(key)}
	}

	def clear() = {
		listeners.clear();
	}
	def add(listner:IKeyListener) : Boolean = {
		return listeners.add(listner);
	}
}
